privacy/ccpa/policy.go

```diff
@@ -25,174 +22,103 @@ func ReadFromRequest(req *openrtb2.BidRequest) (Policy, error) {
 	}
 
 	// Read consent from request.regs.ext
-	if req.Regs != nil && len(req.Regs.Ext) > 0 {
-		var ext openrtb_ext.ExtRegs
-		if err := json.Unmarshal(req.Regs.Ext, &ext); err != nil {
-			return Policy{}, fmt.Errorf("error reading request.regs.ext: %s", err)
-		}
-		consent = ext.USPrivacy
+	err := req.ExtractRegExt()
+	if err != nil {
+		return Policy{}, fmt.Errorf("error reading request.regs.ext: %s", err)
+	}
+	if req.RegExt != nil {
+		consent = req.RegExt.GetUSPrivacy()
 	}
-
 	// Read no sale bidders from request.ext.prebid
-	if len(req.Ext) > 0 {
-		var ext openrtb_ext.ExtRequest
-		if err := json.Unmarshal(req.Ext, &ext); err != nil {
-			return Policy{}, fmt.Errorf("error reading request.ext.prebid: %s", err)
-		}
-		noSaleBidders = ext.Prebid.NoSale
+	err = req.ExtractRequestExt()
+	if err != nil {
+		return Policy{}, fmt.Errorf("error reading request.ext: %s", err)
+	}
+	reqPrebid := req.RequestExt.GetPrebid()
+	if reqPrebid != nil {
+		noSaleBidders = reqPrebid.NoSale
 	}
 
 	return Policy{consent, noSaleBidders}, nil
 }
 
 // Write mutates an OpenRTB bid request with the CCPA regulatory information.
-func (p Policy) Write(req *openrtb2.BidRequest) error {
+func (p Policy) Write(req *openrtb_ext.RequestWrapper) error {
 	if req == nil {
 		return nil
 	}
 
-	regs, err := buildRegs(p.Consent, req.Regs)
+	err := req.ExtractRegExt()
 	if err != nil {
 		return err
 	}
-	ext, err := buildExt(p.NoSaleBidders, req.Ext)
+	buildRegs(p.Consent, req.RegExt)
```

Comment:
`buildRegs` no longer needs to exist. This may be replaced with `regs.SetUSPrivacy("")`. The majority of the logic in this file was for serialization which is now moved to the request wrapper.

Reply:
Thanks, got into the weeds converting everything over to the wrapper I did not notice how much it simplified it in the end.